modules/sql-engine/src/test/java/org/apache/ignite/internal/sql/engine/planner/MapReduceHashAggregatePlannerTest.java

```diff
@@ -318,11 +324,9 @@ public void expandDistinctAggregates() throws Exception {
                         .and(input(isInstanceOf(IgniteReduceHashAggregate.class)
                                 .and(not(hasAggregate()))
                                 .and(hasGroups())
-                                .and(input(isInstanceOf(IgniteExchange.class)
-                                        .and(input(isInstanceOf(IgniteMapHashAggregate.class)
-                                                .and(not(hasAggregate()))
-                                                .and(hasGroups())
-                                        ))
+                                .and(input(isInstanceOf(IgniteMapHashAggregate.class)
+                                        .and(not(hasAggregate()))
+                                        .and(hasGroups())
```

Comment:
I guess, before the change, the plan assumes grouping on affinity columns.
Thus, we can check "DISTINCT" condition on each node locally.

After the fix - plan check the common case, when all the data fetched to single node.
Does it means this case is broken?

Reply:
@AMashenkov Yes, It does. And we should discuss this.

Reply:
As I explained earlier the number of projections/aggregate affects cost of a query plan.  
At the moment MAP/REDUCE use the fixed number of columns (GROUP_ID, LIST, ). After this patch this is no longer holds, add all intermediate expressions are sent as is.

Reply:
Changing the number of columns between affects the cost of `Exchange` operator, because it uses that number (computeSelfCost):
> double rowCount = mq.getRowCount(getInput());
> double bytesPerRow = getRowType().getFieldCount() * IgniteCost.AVERAGE_FIELD_SIZE;

Reply:
Seems, `IgniteMdDistinctRowCount.getDistinctRowCount` is used for calculating aggregate nodes costs.
It's code looks weird. BTW, I have no idea why don't we use default implementation.